### PR TITLE
 [member #2] Create MusicAlbum class

### PR DESCRIPTION
**Pull request by RaphDasilva**
## I did the following

- Created `MusicAlbum` class that inherits from an `Item` class.
- Add a constructor that initializes the three attributes of the `MusicAlbum` object and also calls the constructor of the `Item` class by passing the `publish_date` parameter.
- added some additional behavior specific to music albums, such as checking whether an album can be archived based on whether it is available on Spotify.
- Fix linters

### CODE REVIEW

File: bin/music_album.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require_relative 'item'
+class MusicAlbum < Item
+  attr_accessor :on_spotify, :name
+  attr_reader :publish_date
+
+  def initialize(name, publish_date, on_spotify)
+    super(publish_date)
+    @name = name
```
Comment (anthonymr) [inline bin/music_album.rb:8]: Please, can you check if there is any +name attribute in the UML diagram for this class?

File: bin/music_album.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    @publised_date = publised_date
```
Comment (anthonymr) [inline bin/music_album.rb:9]: the super() constructor is already setting this property, you don't need to set it again. Also, please, notice the typo in this line and in the parameters, it should be publish_date according to the diagram.

File: bin/music_album.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    @on_spotify = on_spotify
+  end
+
+  def can_be_archived?
+    @publish_date < 10.year.ago && @on_spotify
```
Comment (anthonymr) [inline bin/music_album.rb:14]: I tested this code and doesn't work outside `Rails` environment.

Look at this alternative:
```Ruby
  def can_be_archived?
    @publish_date < n_years_ago(10) && @on_spotify
  end

  def n_years_ago(years)
    year = Time.now.year - years
    month = Time.now.month
    day = Time.now.day

    Time.new(year, month, day)
  end
```

**Review by anthonymr — CHANGES_REQUESTED** [review]
Hi @RaphDasilva, good job so far. Please, take a look at the comments below in order to complete the task.

File: bin/music_album.rb (hunk 3 above)
Comment (RaphDasilva) [reply bin/music_album.rb]: Thank you very much, I have done the corrections

**Review by AbdimulhinYussuf3675 — APPROVED** [review]
# STATUS: APPROVED   
 ## Hi @RaphDasilva  
Your project is complete! There is nothing else to say other than... Great job 👏🏾  Congratulations! :tada:
 
 TO HIGHLIGHT:
 
 * Implemented required tasks :white_check_mark:
  * Good Job :+1: :smile: 
 
 Now you can merge this branch and move on 🥇 
 
 ## Optional suggestions

- [x] N/A keep doing the great work :wink: 

 _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
 
Cheers and Happy coding! :clap:  :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me @AbdimulhinYussuf3675 in your question so I can receive the notification.
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**